manifests/params.pp

```diff
@@ -135,6 +135,7 @@
   }
 
   $config_dir = "${etc}/foreman-proxy"
+  $instance_id = foreman_proxy::generate_uuid()
```

Comment:
Two thoughts:

 1) This version would change every time you re-ran the puppet module. Similar to a password, we'd need to cache this to ensure it remained the same. And are there any conditions under which this should change?
 2) Would there be any value in deriving the UUID from the FQDN using https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#fqdn_uuid ? If there was, that should handle the first question

Reply:
1) i was under the impression the installer stored that. Is it not the case? I don't really see a use for changing it once it is set, except if someone had a pre-made vm images which needed adjusting after cloning.

2) Disclaimer: I don't know how UUIDs derived from fqdn work. But in general, if we use something that is derived from a hostname then we can just skip it and do the matching based on hostnames alone, no?

Reply:
https://github.com/voxpupuli/puppet-extlib/blob/master/REFERENCE.md#extlibcache_data

Reply:
Neat, thanks